grakn-engine/src/main/java/ai/grakn/engine/cache/EngineCacheDistributed.java

```diff
@@ -189,21 +195,23 @@ private void addJob(String jobType, String keyspace, String index, ConceptId id)
 
     @Override
     public void clearJobSetResources(String keyspace, String conceptIndex) {
-        cleanJobSet(RESOURCE_JOB, keyspace, conceptIndex);
+        cleanJobSet(RESOURCE_JOB, keyspace, conceptIndex, false);
     }
     @Override
     public void clearJobSetCastings(String keyspace, String conceptIndex) {
-        cleanJobSet(CASTING_JOB, keyspace, conceptIndex);
+        cleanJobSet(CASTING_JOB, keyspace, conceptIndex, false);
     }
-    private void cleanJobSet(String jobType, String keyspace, String index){
+    private void cleanJobSet(String jobType, String keyspace, String index, boolean force){
         String idPath = getPathConceptIds(jobType, keyspace, index);
         Optional<List<String>> ids = getChildrenFromZookeeper(idPath);
 
-        if(ids.isPresent()){
+        if(force && ids.isPresent()){
             ids.get().forEach(id -> deleteJob(jobType, keyspace, index, ConceptId.of(id)));
         }
 
-        deleteObjectFromZookeeper(idPath);
+        if(force || ids.isPresent() && ids.get().isEmpty()){
+            deleteObjectFromZookeeper(idPath);
+        }
```

Comment:
So `cleanJobSet` is used to clear the key of a job. For example the index to be merged. However, it only actually clears this *if* there are no pending jobs under that key. 

*force* lets you clear the jobs regardless if you have any pending or not. And this is needed whenever you clear the graph.